src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -180,6 +182,89 @@ object StructuralVariant {
       graph
     }
 
+    case class SVClique(
+      reads: Set[PairedMappedRead],
+      span: GenomeRange,
+      wiggle: Long
+    )
+
+    def findOneClique(g: PairedReadGraph, maxNormalInsertSize: Int): SVClique = {
+      val edges = g.edges.toList.sortBy(_.weight)
+      val bestEdge = edges.head
+
+      val initPair = bestEdge.nodes.minBy(_.minPos).value
+      var (pairMin, svStart, svEnd, _) = initPair.startsAndStops
+      var wiggle = initPair.insertSize - maxNormalInsertSize - (svEnd - svStart)
+
+      val clique = mutable.Set(initPair)
+
+      val graphSize = g.nodes.length
+
+      var cliqueGrow = true
+      while (cliqueGrow && clique.size < graphSize) {
+        cliqueGrow = false
+        mapFirst(edges, (edge: PairedReadGraph#EdgeT) => {
+          val nodes = edge.nodes.toSeq
+          val Seq(node1, node2) = nodes
+          val node1InClique = clique.contains(node1.value)
+          val node2InClique = clique.contains(node2.value)
+
+          // The edge should have one end in the clique and one end out of it.
+          val node = (node1InClique, node2InClique) match {
+            case (false, true) => Some(node1)
+            case (true, false) => Some(node2)
+            case _             => None
+          }
+
+          // TODO: DELLY caches a list of incompatible vertices
+
+          // The new node should be fully connected with the rest of the clique.
+          node filter { n =>
+            {
+              clique.subsetOf(n.neighbors map { _.value })
+            }
+          } filter { n =>
+            // And it should be compatible with the existing SV.
+            {
+              val (_, nodeGapMin, nodeGapMax, _) = n.value.startsAndStops
```

Comment:
FYI, this is a line that IntelliJ doesn't like: https://youtrack.jetbrains.com/issue/IDEA-143285#